### PR TITLE
fixes for the case list search page and case details page

### PR DESCRIPTION
**Pull request by leepavlouhmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-2569
https://tools.hmcts.net/jira/browse/EUI-2646
https://tools.hmcts.net/jira/browse/EUI-3024

### Change description ###
filter controls' selected values are remembered when you return to the case list search page
label now retained in case heading tab
"any" option for case state is back, and added functionality to search all case states when this is selected.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@hmcts/ccd-case-ui-toolkit",
-  "version": "2.67.6-task-event-completion",
+  "version": "2.68.9-alpha",
```
Comment (rayliangatos) [inline package.json:3]: suggest to add the description between the version num and alpha.

File: src/shared/components/markdown/markdown.html [hunk 2]
```diff
@@ -1 +1 @@
-<div><markdown class="markdown">{{content}}</markdown></div>
\ No newline at end of file
+<div><markdown class="markdown" [innerHTML]="content"></markdown></div>
```
Comment (rayliangatos) [inline src/shared/components/markdown/markdown.html:1]: the STOP sign advise a new line is required

File: src/shared/components/workbasket-filters/workbasket-filters.component.spec.ts [hunk 3]
```diff
@@ -239,12 +239,12 @@ describe('WorkbasketFiltersComponent', () => {
...
-      windowMockService.removeLocalStorage(SAVED_QUERY_PARAM_LOC_STORAGE);
-      component.reset();
-      expect(windowMockService.removeLocalStorage).toHaveBeenCalled();
-    }));
+    // it('should remove localStorage once reset button is clicked', async(() => {
+    //   windowMockService.removeLocalStorage(FORM_GROUP_VAL_LOC_STORAGE);
+    //   windowMockService.removeLocalStorage(SAVED_QUERY_PARAM_LOC_STORAGE);
+    //   component.reset();
+    //   expect(windowMockService.removeLocalStorage).toHaveBeenCalled();
+    // }));
```
Comment (rayliangatos) [inline src/shared/components/workbasket-filters/workbasket-filters.component.spec.ts:247]: tidy these commented code if not needed anymore

File: src/shared/components/workbasket-filters/workbasket-filters.component.spec.ts [hunk 4]
```diff
@@ -497,7 +500,8 @@ describe('WorkbasketFiltersComponent', () => {
         selected: component.selected,
         queryParams: {jurisdiction: JURISDICTION_2.id, 'case-type': DEFAULT_CASE_TYPE.id, 'case-state': DEFAULT_CASE_STATE.id}
       });
-      expect(component.selected.formGroup.value).toEqual(TEST_FORM_GROUP.value);
+      expect(component.selected.formGroup).toEqual(null);
+      // expect(component.selected.formGroup.value).toEqual(TEST_FORM_GROUP.value);
```
Comment (rayliangatos) [inline src/shared/components/workbasket-filters/workbasket-filters.component.spec.ts:504]: tidy up the code please

File: src/shared/components/workbasket-filters/workbasket-filters.component.spec.ts [hunk 5]
```diff
@@ -713,7 +717,8 @@ describe('WorkbasketFiltersComponent', () => {
       let selector = de.query(By.css('#wb-case-state'));
       component.defaults.state_id = CRUD_FILTERED_CASE_TYPES[0].states[0].id;
       expect(selector.nativeElement.selectedIndex).toEqual(0);
-      expect(component.selected.caseState.id).toEqual(CRUD_FILTERED_CASE_TYPES[0].states[0].id);
+      // expect(component.selected.caseState.id).toEqual(CRUD_FILTERED_CASE_TYPES[0].states[0].id);
```
Comment (rayliangatos) [inline src/shared/components/workbasket-filters/workbasket-filters.component.spec.ts:720]: tidy up the code please

File: src/shared/components/workbasket-filters/workbasket-filters.component.ts [hunk 6]
```diff
@@ -84,20 +84,26 @@ export class WorkbasketFiltersComponent implements OnInit {
...
+    }
     this.selected.init = init;
     this.selected.page = 1;
     this.selected.metadataFields = this.getMetadataFields();
     if (init) {
-      this.windowService.setLocalStorage(FORM_GROUP_VAL_LOC_STORAGE, JSON.stringify(this.formGroup.value));
+      if (Object.keys(this.formGroup.controls).length > 0) {
+        this.windowService.setLocalStorage(FORM_GROUP_VAL_LOC_STORAGE, JSON.stringify(this.formGroup.value));
+      }
     }
```
Comment (rayliangatos) [inline src/shared/components/workbasket-filters/workbasket-filters.component.ts:107]: can this if(init) {} block merge with line 86 to 88, another if(init) block there.

File: src/shared/components/workbasket-filters/workbasket-filters.component.ts [hunk 7]
```diff
@@ -106,8 +112,10 @@ export class WorkbasketFiltersComponent implements OnInit {
   reset(): void {
     this.windowService.removeLocalStorage(FORM_GROUP_VAL_LOC_STORAGE);
     this.windowService.removeLocalStorage(SAVED_QUERY_PARAM_LOC_STORAGE);
-    this.resetFieldsWhenNoDefaults();
-    this.onReset.emit(true);
+    setTimeout (() => {
+      this.resetFieldsWhenNoDefaults();
+      this.onReset.emit(true);
+    }, 1000);
```
Comment (rayliangatos) [inline src/shared/components/workbasket-filters/workbasket-filters.component.ts:118]: is the timeout neccessary? if yes can the 1000 reduce to 500 to achieve better performance?

File: src/shared/components/workbasket-filters/workbasket-filters.component.ts [hunk 8]
```diff
@@ -124,7 +132,9 @@ export class WorkbasketFiltersComponent implements OnInit {
       this.selectedJurisdictionCaseTypes = this.selected.jurisdiction.caseTypes.length > 0 ? this.selected.jurisdiction.caseTypes : null;
       this.selected.caseType = this.workbasketDefaults ?
         (this.selectedJurisdictionCaseTypes ? this.selectedJurisdictionCaseTypes[0] : null) : null;
-      this.selected.caseState = this.selected.caseType ? this.selected.caseType.states[0] : null;
+      // this.selected.caseState = this.selected.caseType ? this.selected.caseType.states[0] : null;
```
Comment (rayliangatos) [inline src/shared/components/workbasket-filters/workbasket-filters.component.ts:135]: commented code needs to be removed

File: src/shared/components/workbasket-filters/workbasket-filters.component.ts [hunk 9]
```diff
@@ -124,7 +132,9 @@ export class WorkbasketFiltersComponent implements OnInit {
...
+      this.selected.caseState = null;
+      // this.resetCaseState();
```
Comment (rayliangatos) [inline src/shared/components/workbasket-filters/workbasket-filters.component.ts:137]: commented code needs to be removed

File: src/shared/components/workbasket-filters/workbasket-filters.component.ts [hunk 10]
```diff
@@ -138,7 +148,9 @@ export class WorkbasketFiltersComponent implements OnInit {
   onCaseTypeIdChange(): void {
     if (this.selected.caseType) {
       this.selectedCaseTypeStates = this.sortStates(this.selected.caseType.states);
-      this.selected.caseState = this.selectedCaseTypeStates[0];
+      // this.selected.caseState = this.selectedCaseTypeStates[0];
```
Comment (rayliangatos) [inline src/shared/components/workbasket-filters/workbasket-filters.component.ts:151]: commented code needs to be removed

File: src/shared/components/workbasket-filters/workbasket-filters.component.ts [hunk 11]
```diff
@@ -138,7 +148,9 @@ export class WorkbasketFiltersComponent implements OnInit {
...
+      this.selected.caseState = null;
+      // this.resetCaseState();
```
Comment (rayliangatos) [inline src/shared/components/workbasket-filters/workbasket-filters.component.ts:153]: commented code needs to be removed

File: src/shared/components/workbasket-filters/workbasket-filters.component.html [hunk 12]
```diff
@@ -25,6 +25,7 @@ <h2 class="heading-h2" aria-label="Filters">Filters</h2>
     <select class="form-control form-control-3-4 ccd-dropdown" id="wb-case-state"
             name="state" [(ngModel)]="selected.caseState" [disabled]="isCaseStatesDropdownDisabled()"
             aria-controls="search-result">
+        <option [ngValue]="null">Any</option>
```
Comment (Daniel-Lam) [inline src/shared/components/workbasket-filters/workbasket-filters.component.html:28]: Minor point but is there any reason this `<option>` element is indented more than the one below?

File: src/shared/components/workbasket-filters/workbasket-filters.component.ts [hunk 13]
```diff
@@ -83,21 +83,25 @@ export class WorkbasketFiltersComponent implements OnInit {
...
-    }
     // without explicitly preserving alerts any message on the page
     // would be cleared out because of this initial navigation.
     // The above is only true if no alerts were set prior to loading case list page.
     if (!this.alertService.isPreserveAlerts()) {
       this.alertService.setPreserveAlerts(!this.initialised);
     }
-    this.selected.formGroup = this.formGroup;
+    if (Object.keys(this.formGroup.controls).length === 0) {
+      this.selected.formGroup  = JSON.parse(localStorage.getItem(FORM_GROUP_VAL_LOC_STORAGE));
```
Comment (Daniel-Lam) [inline src/shared/components/workbasket-filters/workbasket-filters.component.ts:93]: Really pedantic but there is some extra whitespace before the `=` sign 🙂

**Review by Daniel-Lam — CHANGES_REQUESTED** [review]
Looks good, @leepavlouhmcts - just a few, very minor formatting changes

File: src/shared/components/workbasket-filters/workbasket-filters.component.ts (hunk 7 above)
Comment (leepavlouhmcts) [reply src/shared/components/workbasket-filters/workbasket-filters.component.ts]: i have changed it to 500. I needed to add setTimeout as i found during testing that local storage would sometimes clear just after this.onRese.emit(true) was called

**Review by Daniel-Lam — APPROVED** [review]

File: src/shared/components/workbasket-filters/workbasket-filters.component.ts (hunk 7 above)
Comment (rayliangatos) [reply src/shared/components/workbasket-filters/workbasket-filters.component.ts]: ok please test whether 500 is not enough and breaking the time sequence.

**Review by rayliangatos — APPROVED** [review]

**Pull request merged**